components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/messaging_reference.adoc

```diff
@@ -0,0 +1,446 @@
+[id="messaging_reference"]
+= Messaging reference
+:description: Comprehensive function and event reference for Starknet messaging.
+
+The Starknet messaging smart contracts include functions and events that you use to implement various flows in a dApp.
+
+For information on the messaging mechanism, see xref:architecture_and_concepts:Network_Architecture/messaging-mechanism.adoc[Messaging mechanism].
+
+The L1 functions, where available, are defined in the following smart contract:
+
+[cols=",,",]
+|===
+|Contract |Description |Functions
+
+|https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol[`StarknetMessaging.sol`] |
+The contract that implements messaging functions.
+
+Interfaces are available through https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/IStarknetMessaging.sol[`IStarknetMessaging.sol`]. a|
+* xref:#cancelL1ToL2Message[`cancelL1ToL2Message`]
+* xref:#consumeMessageFromL2[`consumeMessageFromL2`]
+* xref:#getMaxL1MsgFee[`getMaxL1MsgFee`]
+* xref:#l1ToL2Messages[`l1ToL2Messages`]
+* xref:#sendMessageToL2[`sendMessageToL2`]
+* xref:#startL1ToL2MessageCancellation[`startL1ToL2MessageCancellation`]
+
+|===
+
+The L1 events, where available, are defined in the following smart contract:
+
+[cols=",,",]
+|===
+|Contract |Description |Events
+
+|https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/IStarknetMessagingEvents.sol[`IStarknetMessagingEvents.sol`] |
+The contract that defines messaging events. a|
+* xref:#ConsumedMessageToL1[`ConsumedMessageToL1`]
+* xref:#ConsumedMessageToL2[`ConsumedMessageToL2`]
+* xref:#LogMessageToL1[`LogMessageToL1`]
+* xref:#LogMessageToL2[`LogMessageToL2`]
+* xref:#MessageToL2Canceled[`MessageToL2Canceled`]
+* xref:#MessageToL2CancellationStarted[`MessageToL2CancellationStarted`]
+
+|===
+
+'''
+
+== L1 function reference
+
+Functions are listed in alphabetical order.
+
+'''
+
+[#cancelL1ToL2Message]
+=== `cancelL1ToL2Message`
+
+[discrete]
+==== Description
+
+Cancels an L1 to L2 message. Call this function after calling the `startL1ToL2MessageCancellation()` function. The time between the calls to these two functions must be at least the number of seconds defined by the link:https://github.com/starkware-libs/cairo-lang/blob/caba294d82eeeccc3d86a158adb8ba209bf2d8fc/src/starkware/starknet/solidity/StarknetMessaging.sol#L25[`L1L2_MESSAGE_CANCELLATION_DELAY_TAG`] constant.
```

Comment:
Sort of correct, the constant defines the storage slot the value is actually stored at. To get the real value one should call [this](https://github.com/starkware-libs/cairo-lang/blob/caba294d82eeeccc3d86a158adb8ba209bf2d8fc/src/starkware/starknet/solidity/StarknetMessaging.sol#L59-L61) function

Reply:
So should we add `messageCancellationDelay()` to the reference?

Reply:
I have added `messageCancellationDelay`